### DISCUSSION TITLE
How can I get all images?

### DISCUSSION THREAD

**Discussion by dpriskorn** [earwig/mwparserfromhell#296] [Q&A] [answered]
(I want to count them)

**Comment by earwig** [comment, marked as answer]
Here's a start:

```python
def is_file(link):
    return link.title.lstrip().lower().startswith(('file:', 'image:'))

files = mwparserfromhell.parse(text).filter_wikilinks(matches=is_file)
len(files)
```

There are some considerations:

- This doesn't consider images that are transcluded through a template without the [[bracket]] syntax. A typical example would be an infobox (`{{Infobox person |image=Example.jpg }}` includes `[[File:Example.jpg]]`.)
  - If you want to include these, you might need to expand templates or look at the generated HTML.
  - You could also apply heuristics to template parameter values (e.g. if a template with a name containing "infobox" has a parameter containing "image" whose value is non-empty, that counts as an image—obviously this is pretty fragile).
  - If you look at the HTML or expand all templates, you will also get icons from maintenance templates and similar things that you probably want to exclude.
- Are you dealing with a a non-English wiki? If so, there may be different namespace names for File:/Image: that you should match as well.
- What about non-image files? Could be videos or audio files typically. If you want to exclude these, you might take a look at the file extension, or you might need something more complicated.
